src/emm/clients/git_proxy.py

```diff
@@ -308,10 +313,79 @@ def push_branch_to_remote(self, directory: Optional[Path] = None) -> str:
                 f"in directory '{self._determine_directory(directory)}'"
             )
 
-        args = ["push", "-u", "origin", "HEAD"]
+        args = ["push", "-u", remote, "HEAD"]
         with local.cwd(self._determine_directory(directory)):
             return self.git[args]().strip()
 
+    def current_remote(self, directory: Optional[Path] = None) -> List[str]:
+        """
+        Get the current remote and associate remote repo url.
+
+        :param directory: Directory to execute command at.
+        :return: List of remote names and remote urls.
+        """
+        args = ["config", "--get-regexp", ".url"]
+        with local.cwd(self._determine_directory(directory)):
+            return self.git[args]().strip().split("\n")
+
+    def determine_remote(self, remote_owner: str, directory: Optional[Path]) -> str:
+        """
+        Determine the remote url to push.
+
+        :param remote_owner: The remote user specified.
+        :param directory: Directory to execute command at.
+        :return: The remote url to push.
+        """
+        current_remotes = self.current_remote(directory)
```

Comment:
Why do we need to get all the remotes and parse through them? Why can't we just look up the specified remote directly?